### DISCUSSION TITLE
Material Navigation: Github vs Canvas

### DISCUSSION THREAD

**Discussion by metasean** [USU-ITLS-x265-23-summer/class-repo#16] [Q&A] [answered]
My question is what is the best way to navigate the material in the class? Where should we be spending the bulk of our time? Should we treat Github as our canvas course? (This is new to me so I am just wondering what approach we should take to the learning structure)

_Originally posted by @CJ3315 in https://github.com/USU-ITLS-x265-23-summer/class-repo/discussions/10#discussioncomment-5868106_

**Comment by metasean** [comment, marked as answer]
Wonderful question!

tldr: 
- **Technical Tasks** 
  GitHub `class-repo` & GitHub discussions
- **Instructional tasks**  
  Primarily GitHub discussions, reasonable alternatives in the GitHub `class-repo` will be acceptable
- **Project Tasks (weeks 2, 4, 5, & 6)**  
  I definitely prefer GitHub discussions, but will accept Canvas email conversations
- **Project Tasks (weeks 7+)**  
  - Mostly individual GitHub project repos for actual coding.  
  - I prefer GitHub discussions for non-coding related work, but will absolutely understand if you'd prefer to keep it in your individual project repos.
  - Some of this work could be done via Canvas email - but I _strongly recommend against_ that.
  
- **Weekly Rubrics**  
  Available in Canvas, under the Assignments for that week.
- **Grades**
  - I will be posting grades in Canvas.  
  - My goal is to have the previous week's entries in by each Monday. 
  - Like many of you, I have other obligations.  My primary focus timewise is to ensure y'all are able to continue making progress on your work.  Review of rubric measures is secondary to that.  (I.e. ensuring a discussion question is correctly answered will take priority over grading).  
  - If you don't see credit that you believe should have been given within the following week, send me an email in Canvas, and include (a) the Week #, (b) the Criteria, and (c) a link to the entry you believe you should have met that criteria.
  - For example, if you believe that you "[made] a meaningful addition, modification, clarification, example, or other contribution to the course-repo" about week 2 material (and you actually posted during week 2), and you don't see the corresponding credit for it by the end of week 3, send me a message to the effect of "I believe I met the Week 3 "make a meaningful addition, modification, clarification, example, or other contribution to the course-repo" with http:// link.)

---

Your weekly assignments fall into three categories: technical tasks, instructional tasks, & project tasks.  

You may have noticed, I'm not asking you to demonstrate that you've completed technical tasks.  For the novices among you, you'll probably want to go through these exercises in detail multiple times.  The collective knowledge base and discussions are your way of validating that you've learned what you need to learn, but more importantly, to help each other when you're getting stuck.  For the more experienced among you, you'll still need to go through at least a few of these exercises (see following paragraph for why), but instead of spending significant time doing exercises for material that you're already familiar with, you can spend most of your time clarifying, correcting, providing examples, debugging, etc for the junior folks.  All of _this is done exclusively in GitHub_ (both in the `class-repo` and discussion areas), and roughly a third of your grade is based on your interaction in the knowledge base and discussion.

The instructional tasks are to share and discuss your observations of the different instructional aspects for each week's technical information (unless otherwise specified [1]).  All, or at least the vast majority, of this will be done in GitHub discussions.  It might be useful to aggregate some of this information into the cohort knowledge base (i.e. `class-repo`), which would happen in the GitHub `class-repo`.  From a grading perspective, I'll primarily be focused on your _instructional summaries and related discussions in the GitHub discussion area_. (If I see something that demonstrates comparable results in the `class-repo`, I'll probably count it for grading purposes.)

The project tasks aren't quite as clear-cut.  For all of the initial work (basically weeks 2, 4, 5, & 6 - I'd _prefer_ that we have a running thread in the GitHub discussion area.  This allows you to follow each other's progress, discussions, trains of thought, concerns, ideas, etc.  However, if you'd prefer to complete these through Canvas email conversations, that's permitted.  

Once you're narrowed down to a single project (week 7), then you'll move _primarily to your individual project repo, which will be in GitHub_.  There will still be _aspects that will benefit you and your classmates if done in GitHub discussions_, but some of you will find it easier to keep your project contained within your individual project repo.  (In theory, you could still do it some of it through Canvas email —but it would get messy pretty quickly— so I strongly discourage this method, but if this is what you'd prefer at that point, we can make it work.)

[1]: Week 8 you'll discuss _all_ of the different instructional aspects from the various technical tasks —including this course up to that point!
Be honest!  I have thick skin and much prefer to know how I can improve, than I do false praise or silent disappointment.  Of course, constructive criticism tends to be more useful than just complaints. 

>It's better to light a candle than to curse the darkness.
~ William Lonsdale Watkinson

>Every human being is entitled to courtesy and consideration. >Constructive criticism is not only to be expected but sought.
~ Margaret Chase Smith

**Comment by Skye404** [comment]
This is very helpful! Thanks. I think getting used to Github is going to take some time, so I'm hoping I can find the Discussions and class-repo weekly. Are the discussions always in the same place on a weekly basis? The links in Canvas were helpful for finding this week's materials.

**Reply by metasean** [reply]
Yes - for this course it's best to limit discussions to: https://github.com/orgs/USU-ITLS-x265-23-summer/discussions

You can have a discussion area for your specific project, but I would prefer that you use only the course discussions.

**Reply by Skye404** [reply]
Great, thanks!

**Comment by audreyhki** [comment]
@CJ3315 I believe the majority of the time dedicated to this course will be spent in Github. Canvas can be used to access weekly relevant links and links will redirect you to Github.

**Reply by CJ3315** [reply]
Thank you! Overall I have gotten more comfortable with Github and it is still going to take some time to fully understand things, but I really appreciate your feedback!

**Comment by CJ3315** [comment]
Thank you for providing that information! The more time that I spend in Github the more I get a sense of what it actually is. I am a very hands-on-oriented learner and I prefer to do actions rather than just watch videos/tutorials. The dual functionality of going through the tutorials while actually performing the actions made it so that I learned the material faster and it is more likely to stick. I am overall enjoying this layout although it is challenging and new!
